### PR TITLE
Add View Restriction

### PR DESCRIPTION
**Pull request by jonpas**

### CODE REVIEW

**Comment by Glowbal** [comment]
This has merge conflicts.

**Comment by jonpas** [comment]
Added headless stuff here as well, since all is still heavy WIP, might as well.

File: addons/viewrestriction/functions/fnc_canChangeCamera.sqf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+// Remote control hates switchCamera (control returns to player, camera is left on remotely controlled object/unit), make sure remote controlled units are not impacted
+
+(player == ACE_player) &&
+{ACE_player == cameraOn || vehicle ACE_player == cameraOn} &&
+{!isNull ACE_player} &&
```
Comment (bux578) [inline addons/viewrestriction/functions/fnc_canChangeCamera.sqf]: would it make sense to switch the order of some of the conditions

Comment (jonpas) [inline addons/viewrestriction/functions/fnc_canChangeCamera.sqf]: Probably. :D

**Pull request merged**